media/js/base/core-datalayer.js

```diff
@@ -39,11 +40,13 @@ if (typeof Mozilla == 'undefined') {
                     var element = arguments[i]['gtm.element'];
                     var href = element.href;
 
-                    if (element.hostname === document.location.hostname) {
-                        var path = href.split(element.host)[1];
+                    if (element.hostname === hostname) {
+                        // remove host and locale from internal links
+                        var path = href.replace(/^(?:https?\:\/\/)(?:[^\/])*/, '');
                         var locale = path.split('/')[1];
```

Comment:
If we agree to be more explicit in checking for the presence of a locale, would also be good to add a test.